### PR TITLE
GEOS-7960: Factoring out GeoServerNodeData class from GeoServerNodeInfo.

### PR DESCRIPTION
**Pull request by jdeolive**
Issue link - https://osgeo-org.atlassian.net/browse/GEOS-7960

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/GeoServerNodeData.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
+/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
+ */
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/GeoServerNodeData.java]: Header is usually longer - http://docs.geoserver.org/latest/en/developer/policies/committing.html

Comment (jdeolive) [inline src/main/src/main/java/org/geoserver/GeoServerNodeData.java]: Oops, my bad, will fix.

**Comment by jdeolive** [comment]
@jodygarnett Are you happy with the changes?

**Comment by jodygarnett** [comment]
Go for it; I should ask if there is a need to backport this for your customer, or are you content with the 2.11 release timeframe.

**Comment by jdeolive** [comment]
Thanks @jodygarnett. If possible I would like to backport this one but it is technically an api change... so not sure what the policies are there.

**Comment by jodygarnett** [comment]
Ask on the email list, usually backport is okay for GeoServer (after a one month cooling off period).

**Pull request merged**